### PR TITLE
[PotentialFlow] Adding 3d linear wake conditions 3 of 3

### PR DESCRIPTION
**Pull request by inigolcanalejo**
**Description**
After https://github.com/KratosMultiphysics/Kratos/pull/8785, adding the 3d linear wake conditions for the transonic element.

**Changelog**
- Added 3d linear wake conditions
- Added missing clamped approach
- Updating 2d cpp tests (related to the clamped modification)
- Added 3d cpp tests for all RHS and LHS cases (subsonic, supersonic accelerating, supersonic decelerating, wake, and structure elements)

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp [hunk 1]
```diff
@@ -918,68 +1038,68 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::AssignLeftHandS
...
     const auto& r_geometry = this->GetGeometry();
     for (unsigned int i = 0; i < TNumNodes; ++i)
     {
         // The TE node takes the contribution of the subdivided element and
         // we do not apply the wake condition on the TE node
-        if (r_geometry[i].GetValue(TRAILING_EDGE))
-        {
-            for (unsigned int j = 0; j < TNumNodes; ++j)
-            {
+        if (r_geometry[i].GetValue(TRAILING_EDGE)){
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp]:
```suggestion
        if (r_geometry[i].GetValue(TRAILING_EDGE)) {
```

File: applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp [hunk 2]
```diff
@@ -918,68 +1038,68 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::AssignLeftHandS
...
+            for (unsigned int j = 0; j < TNumNodes; ++j){
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp]:
```suggestion
            for (unsigned int j = 0; j < TNumNodes; ++j) {
```

File: applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp [hunk 3]
```diff
@@ -819,35 +926,48 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateLeftHa
...
     {
-        if (partitions_sign[i] > 0)
+        if (PartitionsSign[i] > 0)
         {
-            noalias(lhs_positive) +=
-                Volumes[i] * density * prod(data.DN_DX, trans(data.DN_DX));
-            noalias(lhs_positive) +=
-                Volumes[i] * 2 * DrhoDu2 * outer_prod(DNV, trans(DNV));
+            noalias(lhs_positive) += Volumes[i] * upper_density * prod(data.DN_DX, trans(data.DN_DX));
+            if( upper_local_velocity_squared < max_velocity_squared){
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp]:
```suggestion
            if(upper_local_velocity_squared < max_velocity_squared) {
```

File: applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp [hunk 4]
```diff
@@ -819,35 +926,48 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::CalculateLeftHa
...
+
         }
         else
         {
-            noalias(lhs_negative) +=
-                Volumes[i] * density * prod(data.DN_DX, trans(data.DN_DX));
-            noalias(lhs_negative) +=
-                Volumes[i] * 2 * DrhoDu2 * outer_prod(DNV, trans(DNV));
+            noalias(lhs_negative) += Volumes[i] * lower_density * prod(data.DN_DX, trans(data.DN_DX));
+            if( lower_local_velocity_squared < max_velocity_squared){
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp]:
```suggestion
            if(lower_local_velocity_squared < max_velocity_squared) {
```

**Review by marcnunezc — DISMISSED** [review]
Ok! Minor format suggestions

**Review by marcnunezc — APPROVED** [review]

**Comment by inigolcanalejo** [comment]
Thanks!

**Pull request merged**